spec/integration/server_spec.rb

```diff
@@ -41,7 +41,7 @@
       it 'can be used for reads' do
         result = view.send(:send_initial_query, server)
         # Driver falls back to the oldest MongoDB protocol
-        expect(result).to be_a(Mongo::Operation::Find::Legacy::Result)
+        expect(result).to be_a(Mongo::Operation::Find::Result)
```

Comment:
This needs to be two tests, each conditioned on server version.

Reply:
This spec only runs on server fcvs >= 3.4, so I don't believe this is a concern for this particular test.

Reply:
Is something preventing it from running on older servers though or was that omitted because the test used to test new servers using the old protocol?

Reply:
The latter. I can update this test to run on all servers and see that it uses the legacy response on old server versions if you think that would improve coverage.